packages/SwingSet/src/kernel/kernel.js

```diff
@@ -746,6 +686,88 @@ export default function buildKernel(
     }
   }
 
+  /*
+   * routeSend(message) figures out where a 'send' event should go. If the
+   * message needs to be queued (it is sent to an unresolved promise without
+   * a pipelining decider), this queues it, and returns null. If the message
+   * goes splat against a dead vat or a non-deliverable resolved promise,
+   * this rejects any result promise, and returns null. Otherwise it returns
+   * the vatID and actual target to which it should be delivered. If the
+   * original target was a promise that has been fulfilled to an object,
+   * this returns that settled object.
+   *
+   * This does not decrement any refcounts. The caller should do that.
+   */
+
+  function routeSendEvent(message) {
+    const { target, msg } = message;
+    const { type } = parseKernelSlot(target);
+    assert(
+      ['object', 'promise'].includes(type),
+      X`unable to send() to slot.type ${type}`,
+    );
+
+    function splat(error) {
+      if (msg.result) {
```

Comment:
Unrelated to the changes in this PR, but it does feel strange to drop errors on the ground if there is nowhere to send them. Have we considered remembering which vat triggered the send, and raise this as a global error there? Probably moot I suppose since live slots automatically creates a promise for the result.

Reply:
Correct, `syscall.send` accepts messages without `.result`, but liveslots has no way to create them, because we have no way to sense that userspace ignored the return from `E().foo()`. @michaelfig had an idea a long while back about putting a WeakRef on the return Promise and watching to see if it got collected before the crank finished (and using that as evidence that we could do a `sendOnly`), but it's fussy enough that we never pursued it. `E.sendOnly(target).method(args)` is still a strong contender though, and I believe the HandledPromise API has a place for it.

I don't really know what to do about the global error handler thing. It's just like a global promise rejection thing. If errors are a normal part of your API (they happen, that's fine, you don't want to make a fuss about it), then it's annoying to have them be announced everywhere (/me casts a stern glance at `marshal`). If they aren't, then even though the sending code doesn't care, the *programmer* should maybe know that something suprising happened, and silently discarding them doesn't help.

For now, I think our stance is "the sending code should care", and we consider it at least a review-worthy problem if someone does `E()` and doesn't `.catch` or `await`. "No error left behind".

From the kernel's point of view, I'm inclined to say that if the vat did `sendOnly`, we'll respect that and not try to work around their decision.

BTW, the only code that does `sendOnly` right now is in devices, like when device-timer tells vat-timer about an alarm going off, and pretty much all of those messages get routed to a built-in vat. So I don't think we have a lot of cases where the kernel dropping an error is going to be provokable by user code.